rules/group.go

```diff
@@ -286,6 +291,10 @@ func (g *Group) run(ctx context.Context) {
 				if missed > 0 {
 					g.metrics.IterationsMissed.WithLabelValues(GroupKey(g.file, g.name)).Add(float64(missed))
 					g.metrics.IterationsScheduled.WithLabelValues(GroupKey(g.file, g.name)).Add(float64(missed))
+					numRulesInGroup := uint64(len(g.rules))
+					if numRulesInGroup > 0 {
+						g.metrics.EvalMissed.WithLabelValues(GroupKey(g.file, g.name)).Add(float64(uint64(missed) * numRulesInGroup))
+					}
```

Comment:
This code seems to be duplicated between the `shouldRestore` and normal cases; could you extract it to a function to avoid the repetition?